packages/radix-ui-themes/src/theme-panel.tsx

```diff
@@ -39,16 +40,16 @@ const ThemePanel = React.forwardRef<ThemePanelImplElement, ThemePanelProps>(
 );
 ThemePanel.displayName = 'ThemePanel';
 
-type ThemePanelImplElement = React.ElementRef<typeof Box>;
-interface ThemePanelImplProps
-  extends React.ComponentPropsWithoutRef<typeof Box>,
-    ThemePanelImplPrivateProps {
-  onAppearanceChange?: (value: Exclude<ThemeOptions['appearance'], 'inherit'>) => void;
-}
+type ThemePanelImplElement = React.ElementRef<'div'>;
+// Narrow to only reference the `asChild` Box prop
+type ThemePanelImplOwnProps = CommonBoxProps & BoxAsChildProps;
```

Comment:
I don't have the context as to why this extends the Box props, lmk if there's a better way to narrow here. I assume we don't want to support the `as` prop with this? If we do we'll need to add it to `Theme` as well.

Reply:
No we don't really as it stands, I'd be happy to remove that. I think we largely consider `ThemePanel` as just drop in with its own style that positions it, slides it in and out of the way etc.

Reply:
Cool that makes sense to me, I can remove it!